### PR TITLE
Remove support for GrSim

### LINKED ISSUE #742: Rename or Change the functionality of grsim_output.SendPrimitives
### Description of the task

grsim_output.sendPrimitives() send more than what the name suggests.

Either we rename the function or change the implementation so that the function only takes primitives as a parameter.

### LINKED ISSUE #982: Make GrSim-Visualizer Run Offline
### Description of the task
Currently, GrSim uses UDP to communicate with visualizer and it uses some sort of wifi/ethernet port of our computers to do that. This means that if we aren't connected to wifi/ethernet, then GrSim cannot communicate with visualizer and it also crashes because it can't find a port, I guess.

This was an issue in Robocup 2019 because we wanted to test plays/tactics on grsim/visualizer on the plane ride over to Sydney.

### Acceptance criteria

- [ ] Make GrSim Visualizer Run Offline

### LINKED ISSUE #1281: Create a new GrSimBackend to output SensorProto and uses ProtoMulticastListener
### Description of the task
Create a new GrSimBackend to output SensorProto and use ProtoMulticastListener instead of NetworkClient. This should be easily swapped in to integrate things in #1135

Note that it must implement the following functionality:
```
void NetworkClient::publishVisionDataWrapper(SSL_WrapperPacket packet)
{
    // We analyze the first 60 packets we receive to find the "real" starting time,
    // because GrSim will sometimes give us garbage packets.
    // The real starting time is the smaller value of the ones we receive
    if (initial_packet_count < 60)
    {
        initial_packet_count++;
        if (packet.has_detection() &&
            packet.detection().t_capture() < last_valid_t_capture)
        {
            last_valid_t_capture = packet.detection().t_capture();
        }
    }
    else
    {
        // We pass all packets without a detection to the logic (since they are likely
        // geometry packet). Packets with detection timestamps are compared to the last
        // valid timestamp to make sure they are close enough before the data is passed
        // along. This ensures we ignore any of the garbage packets grsim sends that
        // are thousands of seconds in the future.
        if (!packet.has_detection())
        {
            publishVisionData(packet);
        }
        else if (std::fabs(packet.detection().t_capture() - last_valid_t_capture) < 100)
        {
            last_valid_t_capture = packet.detection().t_capture();
            publishVisionData(packet);
        }
    }
}
```

### Acceptance criteria

- [ ] Create a new GrSimBackend to output SensorProto and use ProtoMulticastListener instead of NetworkClient 
- [ ] Testing

### Blocked By
 #1283 #1280

### LINKED ISSUE #1416: Remove GrSim
### Description of the task
Once we move to a simulator replacement for GrSim, we can delete it. 
We can also resolve: #982, #742, #1281

### Acceptance criteria

- [ ] Delete estimatePositionAtFutureTime from Ball
- [ ] Delete GrSim
- [ ] :tada:

### Blocked By
#1541 #1536 #1537 #1328

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Remove grsim and a few related hacks we had.

### Testing Done

Code builds, can still use other backends as usual.

### Resolved Issues

resolves #1416
resolves #982
resolves #742
resolves #1281 

### Length Justification

Yeet

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: environment_setup/setup_grsim.sh [hunk 1]
```diff
@@ -1,88 +0,0 @@
-#!/bin/bash
-
-#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#
-# UBC Thunderbots Linux grSim Setup
```
Comment (akhilveeraghanta) [inline environment_setup/setup_grsim.sh:4]: RIP Grsim

**Review by akhilveeraghanta — APPROVED** [review]
LGTM 🔥 🔥 quite the yeetage of grsim

**Review by kelvinkoon — APPROVED** [review]
Press F for grsim.

File: environment_setup/setup_grsim.sh (hunk 1 above)
Comment (MathewMacDougall) [reply environment_setup/setup_grsim.sh:4]: RIP :moyai:

**Comment by MathewMacDougall** [comment]
F

**Comment by kelvinkoon** [comment]
I also like how 3 of the issues involving grsim are resolved just by not having grsim.

**Review by jonathanlew — APPROVED** [review]
:fire: :rocket:

**Review by jordanzliu — APPROVED** [review]
F

**Pull request merged**